Hot reload

mathyba opened the pull request:
annoucing.... hot reload for the configuration!

This required:
- changing docker volume mount to mount entire project directory
- detecting changes in config mtime since last upload
- refactoring config load, connection to intra and handling of httpx client to trigger new signin only when necessary

Side note: using mtime could also enable reload of source files for ultimate dynamic development. Perhaps to implement later on?

mathyba committed:
Change docker volumes to enable config hot reload

Volume needs to be mounted on a directory, so the entire project is now
a volume.
For convenience, the main script has been linked to the project root
directory.

nirae commented on src/slot_checker.py:
```diff
@@ -149,7 +150,7 @@ class ConfigSchema(Schema):
     @post_load
     def create_processing(self, data, **kwargs):
-        return Config(**data)
+        return Config(time.time(), **data)
```
Why not time.time() directly in the Config constructor? in line 106

mathyba commented:
The final decision regarding the config file is to remove possibility to customize its name?

mathyba committed:
Refactor config load and validation schema

Decision has been made to drop the configurable name for the config
file, so there is no more need to parse a config option.

mathyba committed:
Refactor intra connection and authentication

This makes it possible to combine configuration reload and check for valid
connection.

nirae commented:
> The final decision regarding the config file is to remove possibility to customize its name?

Yes!
